cfe_internal/enterprise/federation/federation.cf

```diff
@@ -133,18 +133,53 @@ bundle agent federation_manage_files
         perms => default:mog( "600", "root", "root" );
 
     # TODO: Render these from templates:
-    am_feeder::
-      "$(cfengine_enterprise_federation:config.bin_dir)/dump.sh"
+
+    am_feeder|am_transporter|am_superhub::
+      # TODO: Instrument augments
+      "$(cfengine_enterprise_federation:config.bin_dir)/config.sh"
         create => "true",
+        template_method => "mustache",
+        edit_template => "$(this.promise_dirname)/../../../templates/federated_reporting/config.sh.mustache",
         perms => default:mog( "600", "root", "root" );
-    am_transporter::
-      "$(cfengine_enterprise_federation:config.bin_dir)/transport.sh"
+
+      # TODO: Instrument augments
+      "$(cfengine_enterprise_federation:config.bin_dir)/log.sh"
         create => "true",
+        template_method => "mustache",
+        edit_template => "$(this.promise_dirname)/../../../templates/federated_reporting/log.sh.mustache",
+        template_data => '{ "logger": "log", "logger_args": "", "timestamp_cmd": "date --iso-8601=seconds"}',
+        perms => default:mog( "600", "root", "root" );
```

Comment:
configuring "logger": "log" will cause an infinite loop in log.sh :) Since log() is the function that it defines and then it calls ${CFE_LOGGER}.

```
function log() {
  # $1 -- message to log
  ts="$($CFE_TIMESTAMP_CMD)"
  if [ -n "$ts" ]; then
    "${CFE_LOGGER}" "[$ts] $1"
  else
    "${CFE_LOGGER}" "$1"
  fi
}
```

So likely we want to leave this blank. Remove the "logger" data entry. Might as well remove timestamp_cmd also since what you have there is the default in log.sh.

Reply:
Yeah ok i won't provide ANY template data, that was just an artifact from me starting to look at what should be supported from def.json, but there are so many things that i think we should ignore it for now. Will redact template_data momentarily